intermediate/tcp-async-bi-directional/src/main/java/org/springframework/integration/samples/tcpasyncbi/TcpAsyncBiDirectionalApplication.java

```diff
@@ -0,0 +1,110 @@
+package org.springframework.integration.samples.tcpasyncbi;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.event.EventListener;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.Pollers;
+import org.springframework.integration.dsl.Transformers;
+import org.springframework.integration.ip.IpHeaders;
+import org.springframework.integration.ip.dsl.Tcp;
+import org.springframework.integration.ip.tcp.connection.AbstractClientConnectionFactory;
+import org.springframework.integration.ip.tcp.connection.AbstractServerConnectionFactory;
+import org.springframework.integration.ip.tcp.connection.TcpConnectionCloseEvent;
+import org.springframework.integration.ip.tcp.connection.TcpConnectionOpenEvent;
+import org.springframework.integration.splitter.AbstractMessageSplitter;
+import org.springframework.messaging.Message;
+
+@SpringBootApplication
+public class TcpAsyncBiDirectionalApplication {
+
+	public static void main(String[] args) {
+		SpringApplication.run(TcpAsyncBiDirectionalApplication.class, args);
+	}
+
+	@Bean
+	SampleProperties properties() {
```

Comment:
I don't think this should be a bean since it is a `@ConfigurationProperties`.
What made you to make it an explicit?
Probably has to be in the `@EnableConfigurationProperties` definition...

Reply:
>What made you to make it an explicit?

>Parameter 0 of method Client in org.springframework.integration.samples.tcpasyncbi.ClientPeer required a bean of type 'org.springframework.integration.samples.tcpasyncbi.SampleProperties' that could not be found.

But `@ EnableConfigurationProperties` works too; thanks.

That said; I wasn't wrong 😄 
```
 * Enable support for {@link ConfigurationProperties @ConfigurationProperties} annotated
 * beans. {@code @ConfigurationProperties} beans can be registered in the standard way
 * (for example using {@link Bean @Bean} methods) or, for convenience, can be specified
 * directly on this annotation.
```